libs/server-sent-events/src/client.cpp

```diff
@@ -290,13 +299,13 @@ class FoxyClient : public Client,
 
     void on_read_body(boost::system::error_code ec, std::size_t amount) {
         boost::ignore_unused(amount);
-        if (ec == boost::asio::error::operation_aborted) {
-            logger_("read HTTP response body aborted");
+        if (ec == boost::asio::error::operation_aborted && shutting_down_) {
```

Comment:
If `on_read_body()` can ever be executed from a context that is different from where `do_shutdown()` is executed then we may have a race between modifying & accessing `shutting_down_`.

Reply:
This is a good callout, especially since I had to add the bool back. 

Although the public `async_shutdown` method will `post`'s to the `session_`'s executor, the `run` routine did not - I've fixed that.

Since those are the only two public entrypoints, we should be safe.